### DISCUSSION TITLE
Tracking mode of models

### DISCUSSION THREAD

**Discussion by max-dev-14** [ultralytics/ultralytics#22243] [Docs]
I've got the following situation: I have a dataset consisting of images, that images were got from some videos and the videos, in their turn, were got from a camera (i.e. the dataset is a series of the video fragments). I need to track objects through my dataset. As far as I know I can use `track` method, but it's significant the dataset to be only one video fragment. Is there any method (like `track`) to expand the tracking task to a series of some video  fragments in one whole dataset?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @max-dev-14, thanks for starting the discussion and for using Ultralytics 🚀! This is an automated response to help you get moving quickly—an Ultralytics engineer will also assist you here soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training or tracking ❓ Question, please provide as much information as possible, including:
-Your exact command or Python snippet (e.g., `yolo track ...`)
-Model and version used (e.g., YOLO11/YOLO26, pretrained or custom)
-Environment details (OS, Python, PyTorch, CUDA)
-A small sample of your data layout or a short list of video fragment paths
-Any logs or screenshots that illustrate your current results
-And verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/)

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [Python>=3.8](https://www.python.org/) environment with [PyTorch>=1.8](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

-**Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
-**Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
-**Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
-**Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Reply by glenn-jocher** [reply]
Yes—treat the fragments as one continuous stream: keep a single `YOLO` instance alive and feed frames from each video (or sorted image sequence) with `persist=True`; IDs will carry across file boundaries.

```python
import cv2
from ultralytics import YOLO

model = YOLO("yolo11n.pt")  # or your custom model
video_paths = ["frag1.mp4", "frag2.mp4", "frag3.mp4"]  # ordered chronologically

for vp in video_paths:
    cap = cv2.VideoCapture(vp)
    while cap.isOpened():
        ok, frame = cap.read()
        if not ok:
            break
        result = model.track(frame, persist=True, tracker="botsort.yaml")[0]
        # use result.boxes.id for per-object IDs
    cap.release()
```

Note: passing `source=[v1, v2, ...]` gives each source its own tracker, so drive the loop yourself if you want global IDs. See the tracking guide for `persist=True` usage in the “Persisting Tracks Loop” section in the [Multi‑Object Tracking docs](https://docs.ultralytics.com/modes/track/), the full `track()` API in the [track() reference](https://docs.ultralytics.com/reference/engine/model/), and YAML options like `track_buffer` and ReID in the [tracker configs](https://github.com/ultralytics/ultralytics/tree/main/ultralytics/cfg/trackers).

**Comment by max-dev-14** [comment]
What if the video fragments are absolutly different and there are different objects in the videos? Will IDs be carried across the videos yet?

**Reply by Y-T-G** [reply]
As long as `persist=True`, IDs aren't reset
